### PR TITLE
Add 2 functions Has and Delete to leveldb

### PR DESCRIPTION
**Pull request by WTer**
This change is [Reviewable](https://reviewable.io/reviews/seeleteam/go-seele/39)

### CODE REVIEW

**Comment by lucusfly** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[database/leveldb/leveldb.go, line 57 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/39#-L6PIubeDoYMOFOLun9m:-L6PIubeDoYMOFOLun9n:b43tjvw) ([raw file](https://github.com/seeleteam/go-seele/blob/2ab33c514f54e31f46ab58ddedd5018b8d01013f/database/leveldb/leveldb.go#L57)):*
> ```Go
> 
> func (db *levelDB) HasString(key string) (ret bool, err error) {
> 	return db.db.Has([]byte(key), nil)
> ```

better to use db.Has([]byte(key))

---

*[database/leveldb/leveldb.go, line 65 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/39#-L6PJ28IKxC7msRfECWR:-L6PJ28IKxC7msRfECWS:bhewhlt) ([raw file](https://github.com/seeleteam/go-seele/blob/2ab33c514f54e31f46ab58ddedd5018b8d01013f/database/leveldb/leveldb.go#L65)):*
> ```Go
> 
> func (db *levelDB) DeleteSring(key string) error {
> 	return db.db.Delete([]byte(key), nil)
> ```

better to use db.Delete([]byte(key))

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/39)*

**Comment by lucusfly** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/39)*

**Comment by WTer** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[database/leveldb/leveldb.go, line 57 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/39#-L6PIubeDoYMOFOLun9m:-L6PJmc-lWGzzGx-n8N6:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/2ab33c514f54e31f46ab58ddedd5018b8d01013f/database/leveldb/leveldb.go#L57)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

better to use db.Has([]byte(key))
</blockquote></details>

Done.

---

*[database/leveldb/leveldb.go, line 65 at r1](https://reviewable.io:443/reviews/seeleteam/go-seele/39#-L6PJ28IKxC7msRfECWR:-L6PJoIXqqLl_aFR2DOn:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/2ab33c514f54e31f46ab58ddedd5018b8d01013f/database/leveldb/leveldb.go#L65)):*
<details><summary><i>Previously, lucusfly (lucus) wrote…</i></summary><blockquote>

better to use db.Delete([]byte(key))
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/39)*

**Comment by lucusfly** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/39)*

File: database/leveldb/leveldb_test.go [hunk 1]
```diff
@@ -34,8 +34,34 @@ func Test_LevelDB(t *testing.T) {
 	assert.Equal(t, err, nil)
 	assert.Equal(t, value, "2")
 
+	// check whether key exists
```
Comment (boqiu) [inline database/leveldb/leveldb_test.go]: Recommended to separate to different test cases.

File: database/leveldb/leveldb_test.go [hunk 2]
```diff
@@ -34,8 +34,34 @@ func Test_LevelDB(t *testing.T) {
...
+	exist, err := db.HasString("1")
```
Comment (boqiu) [inline database/leveldb/leveldb_test.go]: Recommend not to write test cases that depend on previous result.

File: database/leveldb/leveldb.go [hunk 3]
```diff
@@ -48,3 +48,19 @@ func (db *levelDB) Put(key []byte, value []byte) error {
 func (db *levelDB) PutString(key string, value string) error {
 	return db.Put([]byte(key), []byte(value))
 }
+
+func (db *levelDB) Has(key []byte) (ret bool, err error) {
```
Comment (boqiu) [inline database/leveldb/leveldb.go]: comment missed?

**Comment by WTer** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[database/leveldb/leveldb_test.go, line 37 at r2](https://beta.reviewable.io/reviews/seeleteam/go-seele/39#-L6PIAZ-23Q_1QTTYG5a-r2-37:-L6P_2kP6eu0Pc5gACnv:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/3d45c7f2479422c434ef910e9dd8e2e73f345c3a/database/leveldb/leveldb_test.go#L37)):*
<details><summary><i>Previously, boqiu wrote…</i></summary><blockquote>

Recommended to separate to different test cases.
</blockquote></details>

Done.

---

*[database/leveldb/leveldb_test.go, line 38 at r2](https://beta.reviewable.io/reviews/seeleteam/go-seele/39#-L6PIAZ-23Q_1QTTYG5a-r2-38:-L6P_5hd1d3YmyXakBSx:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/3d45c7f2479422c434ef910e9dd8e2e73f345c3a/database/leveldb/leveldb_test.go#L38)):*
<details><summary><i>Previously, boqiu wrote…</i></summary><blockquote>

Recommend not to write test cases that depend on previous result.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/seeleteam/go-seele/39)*

**Comment by WTer** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[database/leveldb/leveldb.go, line 52 at r2](https://beta.reviewable.io/reviews/seeleteam/go-seele/39#-L6PIAZ-23Q_1QTTYG5_-r2-52:-L6P_HL_DTK7ycR9cw6B:b-7rfbdu) ([raw file](https://github.com/seeleteam/go-seele/blob/3d45c7f2479422c434ef910e9dd8e2e73f345c3a/database/leveldb/leveldb.go#L52)):*
<details><summary><i>Previously, boqiu wrote…</i></summary><blockquote>

comment missed?
</blockquote></details>

Just wrapper, so we don't add comments for these functions.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/seeleteam/go-seele/39)*

**Comment by WTer** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[database/leveldb/leveldb.go, line 52 at r2](https://beta.reviewable.io/reviews/seeleteam/go-seele/39#-L6PIAZ-23Q_1QTTYG5_-r2-52:-L6QA7i8C6K0JkgcRmuk:b-896fix) ([raw file](https://github.com/seeleteam/go-seele/blob/3d45c7f2479422c434ef910e9dd8e2e73f345c3a/database/leveldb/leveldb.go#L52)):*
<details><summary><i>Previously, WTer wrote…</i></summary><blockquote>

Just wrapper, so we don't add comments for these functions.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/seeleteam/go-seele/39)*

**Comment by lucusfly** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/seeleteam/go-seele/39)*

**Pull request merged**